### ISSUE TITLE
Every single word is misspelled with VSCode Insiders 1.31 and Windows 10

### ISSUE THREAD

**Issue by nickjj** [bartosz-antosik/vscode-spellright#242]
Hi,

Here's my environment. I'm using English on a stock Windows 10 Pro install:

```
Version: 1.31.0-insider (user setup)
Commit: 9fe48f15b1a252a0e9e1ef5975301203baf10b8a
Date: 2019-01-25T08:19:36.277Z
Electron: 3.1.0
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Windows_NT x64 10.0.17763
```

VSCode itself is installed on a different drive than Windows. I'm not sure if this is related but it's probably worth bringing up.

When I open any document with the extension enabled, it flags every single word as misspelled by putting squigglies under everything. It's almost like it's not picking up the dictionary maybe? I have no idea.

Any ideas on how to further troubleshoot this?

**Comment by nickjj** [comment]
I ran CTRL + Shift + P and looked for any spelling related settings. Yep, there was no dictionary selected. After explicitly checking the US dictionary then everything worked.

I'm not sure if this is related to VSCode being on a different drive, but it looks like the extension didn't pick a default dictionary to use.

**Comment by bartosz-antosik** [comment]
Could you please tell me whether this has changed with any recent VSCode upgrade or is it your first install?

**Comment by nickjj** [comment]
It was a fresh install of VSCode on a brand new Windows 10 install where VSCode was never installed on my main Windows drive.

**Comment by nickjj** [comment]
It's also worth pointing out every time I close VSCode and re-open it, I have to pick the dictionary to use. It won't save it to the settings when I select it from the CTRL + Shift + P menu.

**Comment by bartosz-antosik** [comment]
It seems related to the folders that are provided to the extenstion by VSCode. They may be strange in case of "user setup". Shall I have some time I will examine this. No other extension you use has any problem related to configuration files location?

**Comment by nickjj** [comment]
I have no problems with other extensions as far as I know.

My vscode config is available at https://github.com/nickjj/dotfiles/blob/master/.vscode.json

And I'm using these extensions:

```
code \
  --install-extension abusaidm.html-snippets \
  --install-extension ban.spellright \
  --install-extension bierner.markdown-preview-github-styles \
  --install-extension bung87.vscode-gemfile \
  --install-extension coolbear.systemd-unit-file \
  --install-extension CraigMaslowski.erb \
  --install-extension dracula-theme.theme-dracula \
  --install-extension fimars.github-plus-theme-elixir-adapted \
  --install-extension GrapeCity.gc-excelviewer \
  --install-extension Gruntfuggly.todo-tree \
  --install-extension hangxingliu.vscode-nginx-conf-hint \
  --install-extension kirozen.wordcounter \
  --install-extension mechatroner.rainbow-csv \
  --install-extension mikestead.dotenv \
  --install-extension mjmcloug.vscode-elixir \
  --install-extension mrorz.language-gettext \
  --install-extension neilding.language-liquid \
  --install-extension PeterJausovec.vscode-docker \
  --install-extension ryu1kn.partial-diff \
  --install-extension samuelcolvin.jinjahtml \
  --install-extension shanoor.vscode-nginx \
  --install-extension slevesque.vscode-autohotkey \
  --install-extension timonwong.shellcheck \
  --install-extension wholroyd.jinja \
  --install-extension zkirkland.vscode-firstupper
```

**Comment by werunom** [comment]
Hey. Was checking this project's progress and I saw this issue filed. Thought of mentioning here that I have faced this same issue before (probably 2 months back). 

I thought of shifting from the other spell extension to this. But even after installing the dictionary, every word of the file gets highlighted as misspelled. Because of this, I was not able to use this extension and I continued to use the other one. 

Since I didnt have time then, couldnt file the issue. Excuse for that.

**Comment by bartosz-antosik** [comment]
Could anyone please tell me what is your language configuration? What default language is set on Windows?

**Comment by nickjj** [comment]
Mine is `English (United States)`. It's the "default app language". I didn't change anything during or after the Windows installation.

**Comment by werunom** [comment]
My system language is `en_IN` (english-India). I am using vscode in manjaro-linux

**Comment by werunom** [comment]
As i mentioned, I am currently using `code spell checker` extension. In the settings of that, I have set the language as `en_GB`. Since cspell doesnt require separate dictionaries to be installed, I have not downloaded further dictionaries, this current configuration (system having `en_IN` and vscode cspell having `en_GB`) is working fine.

**Comment by bartosz-antosik** [comment]
@werunom Linux is quite different from Windows in the spelling back-end used (it uses Hunspell) so it has to have the dictionaries downloaded. And the dictionaries have to be in UTF-8 format. It is all explained in the README with details. I of course understand your decision about moving to other speller.

**Comment by bartosz-antosik** [comment]
@nickjj I will have a look. Looks like this type of install (user setup, kinda portable mode) may cause the situation. Your setup looks quite legit and the whole situation surprises me.

**Comment by werunom** [comment]
@bartosz-antosik - I would be happy to install the extension again (largely bcz this extension has better support for latex scenarios) and give inputs about my scenario :)  

Do give me couple of days (in the middle of submission). Will update you both on this...

**Comment by nickjj** [comment]
@bartosz-antosik Yeah I think that's the cause too. I bet 99% of Windows people default to installing to their C drive in the standard spot so this issue doesn't come up.

**Comment by bartosz-antosik** [comment]
@werunom Great, please!

**Comment by bartosz-antosik** [comment]
@nickjj Where do you have extensions installed (exact folder)?

**Comment by nickjj** [comment]
- Extensions are installed to `C:\Users\Nick\.vscode-insiders\extensions`
- VSCode itself is installed to `D:\apps\Microsoft VS Code Insiders`

It's really unfortunate VSCode doesn't respect a custom install directory.

**Comment by bartosz-antosik** [comment]
I have tried user setup, with application located somewhere at `E:` drive and extensions in the same directory as yours (`c:\Users\{user-name}\.vscode-insiders\extensions`) and it works. Because my extension uses binary dependencies, I suspected that there may be some security violation (e.g. antivirus or something protests against calling a binary extension) hence the question, but there is not.

Could you tell me whether there are any messages related to SpellRight in `Developer Tools Console` (`Help` -> `Toggle Developer Tools`)?

**Comment by nickjj** [comment]
I have Windows defender and all activate A/Vs disabled by the way.

For the last couple of days the language has been set on startup after I selected it once manually from the menu by the way. At the moment it's "working" and there's no messages added to the dev tools console output related to spell checking.

**Comment by bartosz-antosik** [comment]
@nickjj Thanks for the update. I would love to catch the issue but it seems ephemeric & not directly reproducible. Shall you have any new evidence please report in the thread.

**Comment by nickjj** [comment]
If you have a VM to test on, it might be worth doing a fresh install directly onto a separate drive to see if you can reproduce it.

**Comment by bartosz-antosik** [comment]
@nickjj This may be an idea to try!
